compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java

```diff
@@ -181,14 +181,18 @@ private JsonObject generateRecordType(BRecordType effectiveType, BIntersectionTy
         if (!requiredFields.isEmpty()) {
             typeNode.add("required", requiredFields);
         }
-        // Get record type and set the type name as a property
-        for (BType bType : intersectionType.getConstituentTypes()) {
-            // Does not consider anonymous records
-            if (bType.tag == TypeTags.TYPEREFDESC) {
-                typeNode.addProperty("name", bType.toString().trim());
+        // The tsymbol name is empty for anon intersection type created due to inferred readonly
+        if (!intersectionType.tsymbol.name.value.isEmpty()) {
+            typeNode.addProperty("name", intersectionType.tsymbol.toString().trim());
```

Comment:
Shouldn't this be tsymbol.name.value?

Reply:
`tsymbol.name.value` doesn't return the fully qualified name.